tensorboard/webapp/widgets/custom_modal/custom_modal_test.ts

```diff
@@ -12,183 +12,287 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
```

Comment:
Note: a new test-friendly wrapper component is created to emulate the modal container in the app root.

New tests:
- creates a modal
- runChangeDetection
- closeAll
- cleans up enclosing embeddedView on close

The rest are existing tests that are modified to be compatible with the new test component.